math/fibonacci.c

```diff
@@ -1,23 +1,40 @@
+/**
+ * @file fibonacci.c
+ * @brief Program to print the nth term of the Fibonacci series.
+ * @details Fibonacci series generally starts from 0 and 1. Every next term in
+ *the series is equal to the sum of the two preceding terms.
+ **/
+
 #include <stdio.h>
 
-// Fibonnacci function
-int fib(int number)
-{
-    if (number == 1 || number == 2)
-        return 1;
-    else
-        return fib(number - 1) + fib(number - 2);
-}
+unsigned fib(int);
 
+/* Driver Code */
 int main()
 {
     int number;
 
-    // Asks for the number that is in n position in Fibonnacci sequence
-    printf("Number: ");
+    // Getting n
+    printf("Enter n to find nth fibonacci element: ");
     scanf("%d", &number);
 
-    printf("%d \n", fib(number));
-
+    printf("%u\n", fib(number));
     return 0;
-}
\ No newline at end of file
+}
+
+/**
+ * @brief Determines the nth Fibonacci term
+ * @param number - n in "nth term" and it can't be negative
+ * @return nth term in unsigned type
+ * @warning Only till 47th and 48th fibonacci element can be stored in 'int' and
+ * 'unsigned' respectively (takes more than 20 seconds to print)
+ */
+unsigned fib(int number)
+{
```

Comment:
Maybe you should make `number` `unsigned` too, or a check like this would be great
```suggestion
{
    if (number < 1) {
        fprintf(stderr,"Illegal Argument!");
        exit(EXIT_FAILURE);
    }
```

Reply:
May I close this PR and add another PR with the corrections mentioned here?

Reply:
That's not necessary, if you haven't noticed, you should know that the entire PR will be squashed into a single commit after merge, just push your update to your fork.